go/merkletree2/proof_verifier.go

```diff
@@ -100,3 +103,180 @@ func (m *MerkleProofVerifier) VerifyInclusionProof(ctx logger.ContextInterface,
 	// Success!
 	return nil
 }
+
+func (m *MerkleProofVerifier) computeSkipsHashForSeqno(s Seqno, skipsMap map[Seqno]Hash) (Hash, error) {
+	skipSeqnos := SkipPointersForSeqno(s)
+	skips := make([]Hash, len(skipSeqnos))
+	for i, s := range skipSeqnos {
+		skip, found := skipsMap[s]
+		if !found {
+			return nil, fmt.Errorf("the skipsMap in the proof does not contain necessary hash of seqno %v ", s)
+		}
+		skips[i] = skip
+	}
+	_, hash, err := m.cfg.Encoder.EncodeAndHashGeneric(skips)
+	if err != nil {
+		return nil, fmt.Errorf("Error encoding %+v: %v", skips, err)
+	}
+
+	return hash, nil
+}
+
+// computeFinalSkipPointersHashFromPath recomputes the SkipPointersHash for all
+// the seqnos on the SkipPointersPath(initialSeqno, finalSeqno), using the
+// information contained in the proof and returns the last one, as well as a
+// boolean indicating wether the proof is a full proof or is compressed (i.e. it
+// is part of a MerkleInclusionExtensionProof).
+func (m *MerkleProofVerifier) computeFinalSkipPointersHashFromPath(ctx logger.ContextInterface, proof *MerkleExtensionProof, initialSeqno Seqno, initialRootHash Hash, finalSeqno Seqno) (h Hash, isPartOfIncExtProof bool, err error) {
+	rootHashMap := make(map[Seqno]Hash)
+	rootHashes, err := ComputeRootHashesNeededInExtensionProof(initialSeqno, finalSeqno)
```

Comment:
You're doing the sort in `ComputeRootHashesNeededInExtensionProof` and then throwing the sort order away when you turn it into a hash. Not the biggest deal

Reply:
Good catch. The sort is 1) unnecessary for how I use the output and 2) actually a no-op as the array is already being generated in sorted order.